components/speedreader/rust/lib/src/readability/src/extractor.rs

```diff
@@ -285,11 +287,24 @@ pub fn extract_dom(
 
             // Our CSS formats based on id="article".
             dom::set_attr("id", "article", body.clone(), true);
+            dom::set_attr("hidden", "true", body.clone(), true);
             body.to_string()
         }
         _ => top_candidate.to_string(),
     };
 
+    for node in meta.preserved_meta.iter() {
+        if let Some(data) = node.as_element() {
+            let attributes = data.attributes.borrow();
+
+            let mut val: String = String::from("<meta ");
+            for attr in attributes.map.iter() {
+                val += &format!(" {}=\"{}\" ", attr.0.local, attr.1.value.to_string());
```

Comment:
FWIW the `.to_string()` here isn't necessary. `Tendril<UTF8>` implements the required methods to be used as a format argument.